src/api-v1.js

```diff
@@ -580,6 +580,94 @@ api.declare({
 
 api.declare({
   method: 'put',
+  route: '/ami-set/:amiSetId',
+  name: 'createAmiSet',
+  deferAuth: true,
+  scopes: [['aws-provisioner:manage-ami-set:<amiSetId>']],
+  title: 'Create new AMI Set',
+  stability:  base.API.stability.stable,
+  description: [
+    'Create an AMI Set. An AMI Set is a collection of AMIs with a single name.',
+  ].join('\n'),
+}, async function (req, res) {
+  let input = req.body;
+  let AmiSetId = req.params.id;
+
+  input.lastModified = new Date();
+
+  // Authenticate request with parameterized scope
+  if (!req.satisfies({amiSet: AmiSetId})) {
+    return;
+  }
+
+  // Create amiSet
+  let amiSetId;
+  try {
+    amiSetId = await this.AmiSet.create(amiSet, input);
+  } catch (err) {
+    // We only catch EntityAlreadyExists errors
+    if (!err || err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+    amiSetId = await this.AmiSet.load({AmiSetId});
+
+    // Check if it matches the existing amiSet
+    let match = [
+      'id',
+      'amis',
+    ].every((key) => {
+      return _.isEqual(amiSetId[key], input[key]);
+    });
+
+    // If we don't have a match we return 409, otherwise we continue as this is
+    // is an idempotent operation.
+    if (!match) {
+      res.status(409).json({
+        error: 'AMI Set already exists with different definition',
+      });
+      return;
+    }
+  }
+
+});
+
+api.declare({
+  method: 'delete',
+  route: '/ami-set/:amiSetId',
+  name: 'removeAmiSet',
+  deferAuth: true,
+  scopes: [['aws-provisioner:manage-ami-set:<amiSetId>']],
+  title: 'Delete AMI Set',
+  stability:  base.API.stability.stable,
+  description: [
+    'Delete an AMI Set.',
+  ].join('\n'),
+}, async function (req, res) {
+  let that = this;
+  let amiSet = req.params.amiSetId;
+
+  try {
+    await this.amiSet.remove({amiSet: amiSetId}, true);
```

Comment:
Remember that you're passing in the AmiSet entity class as a context variable to this function in `src/main.js`.  This name here needs to be the name of the context object key, since what we do is take the 'context' object and assign all those values as properties on the `this` in this scope.  In other words, you're looking for `this.AmiSet.remove`.